src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.Unix.cs

```diff
@@ -0,0 +1,115 @@
+                    if (dirHandle != IntPtr.Zero)
+                    {
+                        try
+                        {
+                            Interop.Sys.DirectoryEntry dirent;
+                            while (Interop.Sys.ReadDirR(dirHandle, dirBufferPtr, bufferSize, out dirent) == 0)
+                            {
+                                if (dirent.InodeType != Interop.Sys.NodeType.DT_DIR)
+                                    continue;
+
+                                Span<char> nameBuffer = stackalloc char[Interop.Sys.DirectoryEntry.NameBufferSize];
+                                ReadOnlySpan<char> entryName = dirent.GetName(nameBuffer);
+
+                                if (cultureName.Length != entryName.Length)
+                                    continue;
+
+                                string entryNameString = entryName.ToString();
+
+                                if (!cultureName.Equals(entryNameString, StringComparison.InvariantCultureIgnoreCase))
+                                    continue;
+
+                                string assemblyPath = $"{parentDirectory}/{entryNameString}/{assembly}.dll";
+
+                                if (Internal.IO.File.InternalExists(assemblyPath))
+                                    return assemblyPath;
+                            }
+                        }
+                        finally
+                        {
+                            if (dirHandle != IntPtr.Zero)
+                                Interop.Sys.CloseDir(dirHandle);
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                if (dirBuffer != null)
+                    ArrayPool<byte>.Shared.Return(dirBuffer);
+            }
+            return null;
+        }
+
+        static private Assembly? ResolveSatelliteAssembly(AssemblyLoadContext alc, AssemblyName assemblyName)
+        {
+            string? cultureName = assemblyName.CultureName;
+
+            if (cultureName == null || cultureName.Length == 0)
+                return null;
+
+            if (assemblyName.Name == null)
+                return null;
+
+            AssemblyName parentAssemblyName = new AssemblyName(assemblyName.Name);
+
+            Assembly? parentAssembly = alc.LoadFromAssemblyName(parentAssemblyName);
+
+            if (parentAssembly == null)
+                return null;
+
+            AssemblyLoadContext? parentAlc = GetLoadContext(parentAssembly);
+
+            if (parentAlc == null)
+                return null;
+
+            string parentDirectory = Path.GetDirectoryName(parentAssembly.Location)!;
+
+            string assemblyPath = $"{parentDirectory}/{cultureName}/{assemblyName.Name}.dll";
+            if (Internal.IO.File.InternalExists(assemblyPath))
+            {
+                Assembly satelliteAssembly = parentAlc.LoadFromAssemblyPath(assemblyPath);
```

Comment:
>Assembly satelliteAssembly [](start = 16, length = 26)

nit: Maybe you can define 

Assembly satelliteAssembly = null; 

before the if statement, then don't check for null values inside the if at all and let the method return satelliteAssembly.